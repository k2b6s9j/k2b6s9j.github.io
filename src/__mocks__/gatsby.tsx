import * as React from "react";
const gatsby = jest.requireActual("gatsby");
import { graphql } from "graphql";

type MockedGraphQLReturn = Promise<any>;

const mockedGraphQLQuery = (query: string): MockedGraphQLReturn =>
  graphql(undefined, query);

interface StaticQueryHOCProps {
  query: MockedGraphQLReturn;
  render?: (data: object) => React.ReactElement;
  children?: (data: object) => React.ReactElement;
}

const mockedStaticQueryHOC = (
  props: StaticQueryHOCProps
): React.ReactElement => {
  if (!props.children || !props.render) {
    throw new Error("Render prop must be provided!");
  }

  return (
    (props.children && props.children({})) ||
    (props.render && props.render({})) || <div>Error!</div>
  );
};

const mockedStaticQueryHook = (data: MockedGraphQLReturn): object => ({});

module.exports = {
  ...gatsby,
  graphql: jest.fn(mockedGraphQLQuery),
  Link: jest.fn(
    ({
      // these props are invalid for an `a` tag
      /* eslint-disable @typescript-eslint/no-unused-vars */
      activeClassName,
      activeStyle,
      getProps,
      innerRef,
      partiallyActive,
      ref,
      replace,
      /* eslint-enable @typescript-eslint/no-unused-vars */
      to,
      ...rest
    }): React.ReactElement => <a {...rest} href={to} />
  ),
  StaticQuery: jest.fn(mockedStaticQueryHOC),
  useStaticQuery: jest.fn(mockedStaticQueryHook)
};
