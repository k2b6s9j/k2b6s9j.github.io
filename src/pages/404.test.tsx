import * as React from "react";
import renderer from "react-test-renderer";
import { useStaticQuery } from "gatsby";
import Page from "./404";

beforeEach(() => {
  (useStaticQuery as jest.Mock).mockImplementationOnce(() => ({
    backdrop: {
      childImageSharp: {
        fluid: {}
      }
    },
    file: {
      childImageSharp: {
        fixed: {
          base64: ""
        }
      }
    },
    site: {
      siteMetadata: {
        title: ""
      }
    }
  }));
});

describe("404 Page", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(<Page data={{ desktop: { childImageSharp: { fluid: {} } } }} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
