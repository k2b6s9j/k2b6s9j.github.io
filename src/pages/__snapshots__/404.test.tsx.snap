// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`404 Page renders correctly 1`] = `
.emotion-1 {
  max-height: 100vh;
  max-width: 100vw;
}

.emotion-0 {
  height: 100vh;
  width: 100vw;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  text-align: center;
  -webkit-backdrop-filter: blur(5px);
  backdrop-filter: blur(5px);
}

<head>
  Object {}
  Object {
    "lang": "en",
  }
  <meta
    content="Kepler Sticka-Jones"
    data-react-helmet={true}
    name="author"
  />
  <meta
    data-react-helmet={true}
    name="description"
  />
  <meta
    content="#409040"
    data-react-helmet={true}
    name="theme-color"
  />
  <meta
    content="404! | "
    data-react-helmet={true}
    property="og:title"
  />
  <meta
    content="website"
    data-react-helmet={true}
    property="og:type"
  />
  <meta
    data-react-helmet={true}
    property="og:url"
  />
  <meta
    content=""
    data-react-helmet={true}
    property="og:image"
  />
  <meta
    content="image/jpeg"
    data-react-helmet={true}
    property="og:image:type"
  />
  <meta
    content="480"
    data-react-helmet={true}
    property="og:image:width"
  />
  <meta
    content="480"
    data-react-helmet={true}
    property="og:image:height"
  />
  <meta
    data-react-helmet={true}
    property="og:description"
  />
  <meta
    content="summary"
    data-react-helmet={true}
    name="twitter:card"
  />
  <meta
    content="@test"
    data-react-helmet={true}
    name="twitter:site"
  />
  <meta
    content="@test"
    data-react-helmet={true}
    name="twitter:creator"
  />
  <script
    data-react-helmet={true}
    type="application/ld+json"
  >
    Object {
    "@context": "http://www.schema.org",
    "@type": "WebPage",
    "name": "404! | Kepler Sticka-Jones",
  }
  </script>
  <title
    data-react-helmet={true}
  >
    404! | 
  </title>
</head>

<main>
  <section
    className="emotion-1 gbi-00000000000-0000000000000000000000 gatsby-image-wrapper"
    style={
      Object {
        "backgroundPosition": "center",
        "backgroundRepeat": "no-repeat",
        "backgroundSize": "cover",
        "opacity": 0.99,
        "position": "relative",
      }
    }
  >
    <style
      dangerouslySetInnerHTML={
        Object {
          "__html": "
          .emotion-1.gbi-00000000000-0000000000000000000000:before,
          .emotion-1.gbi-00000000000-0000000000000000000000:after {
            content: '';
            display: block;
            position: absolute;
            width: 100%;
            height: 100%;
            top: 0;
            left: 0;
            background-color: #040e18;
            transition: opacity 0.5s ease 0.25s;
            background-position: center;
background-repeat: no-repeat;
background-size: cover;

          }
          .emotion-1.gbi-00000000000-0000000000000000000000:before {
            z-index: -100;
            
            
            opacity: 0; 
          }
          .emotion-1.gbi-00000000000-0000000000000000000000:after {
            z-index: -101;
            
            
            
          }
        ",
        }
      }
    />
    <noscript>
      <style
        dangerouslySetInnerHTML={
          Object {
            "__html": "
          .emotion-1.gbi-00000000000-0000000000000000000000:before {
            opacity: 1;
            background-image: ;
          }",
          }
        }
      />
    </noscript>
    <div
      className="emotion-0"
    >
      <div>
        <h1>
          404!
        </h1>
        <span>
          Maybe, go back to the 
          <a
            href="/"
          >
            home page
          </a>
          ?
        </span>
      </div>
    </div>
  </section>
</main>
`;
